rust/azure_iot_operations_connector/examples/filemount_azure_device_registry_observation.rs

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! This example demonstrates how to use the Azure Device Registry (ADR) file mount
+//! observation feature of the Azure IoT Operations Connector.
+//!
+//! To use the example, set the `ADR_RESOURCES_NAME_MOUNT_PATH` environment variable to the
+//! directory where the file mount will be created. The example will create
```

Comment:
Ideally we wouldn't have to mock this for the sample, but this is very convenient for now